code/modules/mob/living/carbon/xenomorph/Powers.dm

```diff
@@ -2365,23 +2367,14 @@
 
 	addtimer(CALLBACK(src, .drone_sting_cooldown), DRONE_STING_COOLDOWN)
 
-	drone_recurring_injection(H)
+	recurring_injection(H, "xeno_growthtoxin")
 
 /mob/living/carbon/Xenomorph/Drone/proc/drone_sting_cooldown()
 	playsound(loc, 'sound/voice/alien_drool1.ogg', 50, 1)
 	to_chat(src, "<span class='xenodanger'>You feel your growth toxin glands refill. You can use Growth Sting again.</span>")
 	update_action_button_icons()
 
-/mob/living/carbon/Xenomorph/Drone/proc/drone_recurring_injection(mob/living/H, count = 2)
-	//set waitfor = FALSE
-	while(count)
-		if(!Adjacent(H) || stagger)
-			return FALSE
-		face_atom(H)
-		if(!do_after(src, DRONE_STING_CHANNEL_TIME, TRUE, 5, BUSY_ICON_HOSTILE))
-			break
-		animation_attack_on(H)
-		playsound(H, pick('sound/voice/alien_drool1.ogg', 'sound/voice/alien_drool2.ogg'), 15, 1)
-		H.reagents.add_reagent("xeno_growthtoxin", DRONE_STING_AMOUNT_RECURRING, null, 300, FALSE, FALSE, TRUE) //Caps the amount injected by the overdose limit
-		count--
-	return
+/mob/living/carbon/Xenomorph/proc/can_sting(mob/living/H)
+	if(!istype(H) || isxeno(H) || iscyborg(H) || H.stat == DEAD)
```

Comment:
The idea here is to avoid the `istype` checks, so instead of using `if(!can_sting(H))` you use `if(!H.can_sting())`
In that sense you do stuff like
```
/mob/proc/can_sting()
	return FALSE

/mob/living/carbon/monkeycan_sting()
	return TRUE

/mob/living/carbon/human/can_sting()
	if(stat != DEAD)
		return TRUE

/mob/living/carbon/human/species/machine/can_sting()
	return FALSE

/mob/living/carbon/human/species/synthetic/can_sting()
	return FALSE
```